content/rust/concepts/operators/operators.md

```diff
@@ -0,0 +1,238 @@
+| | `&mut` | Mutable borrow |
+| 13 | `as` | Type casting keyword |
+| | `:` | Multiple uses |
+| 12 | `* / %` | Multiplication, Division, Remainder |
+| 11 | `+ -` | Addition, Subtraction |
+| 10 | `<< >>` | Bitwise left shift and right shift |
+| 9 | `&` | Bitwise or Logical AND |
+| 8 | `^` | Bitwise or Logical XOR |
+| 7 | `|` | Bitwise or Logical OR |
+| 6 | `== !==` | Equality, Inequality |
+| | `< <= > >=` | Less than, Less than or equal, Greater than, Greater than or eqaul |
+| 5 | `&&` | Logical AND |
+| 4 | `||` | Logical OR |
+| 3 | `.. ..=` | Range literal, Assignment by range literal |
+| 2 | `=` | Direct assignment |
+| | `+= -= *= /= %=` | Compound assignment by sum, difference, product, quotient and remainder |
+| | `<<= >>=` | Compound assignment by Bitwise left shift and right shift |
+| | `&= ^=` `|=` | Compound assignment by Bitwise AND, XOR and OR |
+| 1 | `return` | Return statement |
+| | `break` | Break statement |
+
+Operators with higher precedence are evaluated before operators with lower precedence but when operators have the same precedence, the associativity of the operators determines the order in which the operation is performed.
+Below is a table showing the associativity of rust operators:
+
+| Operator | Description | Associativity |
+| --- | --- | --- |
+| `Field expressions` | Expressions | Left to Right | 
+| `as` | 	Type casting keyword |
+| `:` | Operator (multiple uses) |
+| `* / %` | Multiplication, Division, Remainder |
+| `<< >>` | Bitwise left shift and right shift |
+| `&` | Bitwise or Logical AND |
+| `^` | Bitwise or Logical XOR |
+| `&&` | Logical AND |
+| `=` | Direct assignment | Right to Left |
+| `+= -= *= /= %=` | Compound assignment by sum, difference, product, quotient and remainder |
+| `<<= >>=` | Compound assignment by Bitwise left shift and right shift |
+| `&= ^=` `|=` | Compound assignment by Bitwise AND, XOR and OR |
+
+## Types of Operators
+
+| Category | Operator | Description |
+| --- | --- | --- |
+| Arithmetic Operators | `+` | Addition |
+| | `-` | Subtraction |
+| | `*` | Multiplication |
+| | `/` | Division |
+| | `%` | Modulo |
+| Comparison Operators | `==` | Is equal to |
+| | `!=` | Not equal to |
+| | `>` | Greater than |
+| | `<` | Less than |
+| | `>=` | Greater than or equal to |
+| | `<=` | Less than or equal to |
+| Logical Operators | `&&` | Logical AND |
+| | `||` | Logical OR |
+| | `!` | Logical NOT |
+| Bitwise Operators | `&` | Bitwise AND |
+| | `|` | Bitwise OR |
+| | `^` | Bitwise XOR |
+| | `!` | Bitwise NOT |
+| | `<<` | Left shift |
+| | `>>` | Right shift |
+| Compound Assignment Operators | `+=` | Addition assignment |
+| | `-=` | Subtraction assignment |
+| | `*=` | Multiplication assignment |
+| | `/=` | Division assignment |
+| | `%=` | Modulo assignment |
+
+## Example
+
+Here is a simple code block that demonstrates the use of various operators in rust:
+
+``` rust
+fn main() {
+  let x = 8;
+  let y = 4;
+  let z = 10
```

Comment:
```suggestion
  let mut x = 8;
  let mut y = 4;
  let mut z = 10;
```
You need to make this change in order for the code to be runnable. You left it out, please make this change.